scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -937,9 +937,17 @@ class TypedPipeInst[T] private[scalding] (@transient inpipe: Pipe,
       // for us. So unwind until you hit the first filter, snapshot,
       // then apply the unwound functions
       case Some((tap, fields, Converter(conv))) =>
-        Execution.from(new Iterable[T] {
-          def iterator = m.openForRead(conf, tap).asScala.map(tup => conv(tup.selectEntry(fields)))
-        })
+        /*
+           * We create a Stream here so we are lazy with construction, but we don't read twice.
+           */
+        val tryStream = scala.util.Try(m.openForRead(conf, tap)
+          .asScala
+          .map(tup => conv(tup.selectEntry(fields)))
+          .toStream)
```

Comment:
@ianoc Yes, if it's 100s of part files, being read over HDFS by 10,000 mappers, that's gonna suck - but if that becomes a single distributed cache file that they're each reading from local disk, not so bad. (And it's not like the reads magically go away just because it's in the jobconf, right? You're just adding serialization overhead).

In practice right now, I'm doing stuff in the 100MB range. So serializing into JobConf will probably work. But more generally, this feels like a primitive you might want to implement hashJoin-like things with; are we comfortable with those always working this way? Given that large JobConfs go out over the distcache anyway, does that make any other use of the distcache redundant?